tawk_to.admin.inc

```diff
@@ -1,260 +1,192 @@
-                    action     : 'set_visibility',
-                    ajax       : true,
-                    page_id    : jQuery('input[name="page_id"]').val(),
-                    widget_id  : jQuery('input[name="widget_id"]').val(),
-                    options    : jQuery(this).serialize()
-                }, function(r) {
-                    if(r.success) {
-                        document.body.scrollTop = 0; // For Safari
-                        document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
-                        jQuery('#optionsSuccessMessage').toggle().delay(10000).fadeOut();
-                    }
-                });
+  // include urls config
+  $show_oncustom_default_value = '';
+  if (!is_null($display_opts) && $display_opts->show_oncustom) {
+    $show_oncustom_default_value = format_multiline_url_paths($display_opts->show_oncustom);
+  }
+  $form['visibility_settings']['show_oncustom'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Show on pages:'),
+    '#description' => t('Add URLs to pages in which you would like to show the widget.<br>
+    Put each URL in a new line and just include the the leading \'/\' and page URL (e.g. /about)'),
+    '#prefix' => '<div class="div_show_specific">',
+    '#suffix' => '</div>',
+    '#id' => t('show_oncustom'),
+    '#default_value' =>  $show_oncustom_default_value
+  );
 
-                // stop the form from submitting the normal way and refreshing the page
-                event.preventDefault();
-            });
-        });
-    </script>
-<?php
-    $markup = ob_get_contents();
-    ob_end_clean();
-    return $markup;
-}
\ No newline at end of file
+  // privacy settings
+  $form['privacy_settings'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Privacy Settings'),
+    '#description' => t('Define whether or not user information can be used')
+  );
+
+  // enable visitor recognition config
+  $enable_visitor_recognition_attr = array();
+  if (!is_null($display_opts) && $display_opts->enable_visitor_recognition) {
+    $enable_visitor_recognition_attr['checked'] = true;
+  }
+  $form['privacy_settings']['enable_visitor_recognition'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Enable visitor recognition'),
+    '#description' => t('If selected, name and email address from logged in users will be used to identify the user to you when a chat comes in via tawk.to'),
+    '#id' => t('enable_visitor_recognition'),
+    '#attributes' => $enable_visitor_recognition_attr
+  );
+
+  // submit button
+  $form['actions']['#type'] = 'actions';
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save configuration'),
+  );
+
+  $form['#submit'][] = 'tawk_to_visibility_opts_form_submit';
+
+  return $form;
+}
+
+/**
+ * tawk.to visibility options form submit handler
+ */
+function tawk_to_visibility_opts_form_submit($form, &$form_state) {
+  $jsonOpts = array(
```

Comment:
PHP variable names for arrays should be $json_opts. Camel-case variable names are usually reserved for objects.